### PR TITLE
slight refactoring of ARP work 

### PR DESCRIPTION
**Pull request by si74**

### CODE REVIEW

File: arp.go [hunk 1]
```diff
@@ -201,3 +209,97 @@ func (q *ARPQueue) IterateAndRun(ip string, fn func([]byte) error) {
...
+	binary.BigEndian.PutUint16(frameBuf[22:24], HlenPlen)
+
+	// ARP operation, request is 1
+	binary.BigEndian.PutUint16(frameBuf[24:26], OperationRequest)
+
+	// ARP Sender hardware address (SHA)
+	copy(frameBuf[26:32], gMAC[0:6])
+
+	// ARP Sender protocol address (SPA)
+	spa := net.ParseIP(switchIP).To4()
```
Comment (si74) [inline arp.go:262]: @nicboul I was curious about this - why exactly are we using the switch IP as the source IP for ARPs?

File: arp.go [hunk 2]
```diff
@@ -210,34 +210,32 @@ func (q *ARPQueue) IterateAndRun(ip string, fn func([]byte) error) {
...
 	if err != nil {
 		return nil, fmt.Errorf("unable to add waiting ARP entry: %v", err)
 	}
 
-	// TODO(sneha): use queueing later.
-	//arpQueue.Add(dstIP.String(), frameBuf[0:4+14+n])
+	// Make space for nv header + ethernet header + ARP request
+	frameBuf := make([]byte, 4+14+28)
+
+	// nvHeader length value (the 2 bytes length field doesn't count so it's 2, not 4 byteshhh
```
Comment (si74) [inline arp.go:225]: @nicboul can you explain this is a bit more? why is it that the 2 bytes length field doesn't count?

File: arp.go [hunk 3]
```diff
@@ -210,34 +210,32 @@ func (q *ARPQueue) IterateAndRun(ip string, fn func([]byte) error) {
...
-	copy(frameBuf[10:16], gMAC[0:6])
-
-	// The destination MAC is unknown, store packet and send an ARP request.
+	// FIXME this function should be .Upsert() (Add or Update the entry)
 	err := arpTable.Add(dstIP)
 	if err != nil {
 		return nil, fmt.Errorf("unable to add waiting ARP entry: %v", err)
 	}
 
-	// TODO(sneha): use queueing later.
```
Comment (si74) [inline arp.go]: Is there a place within the code that would be appropriate to drop the TODO - maybe not withing `GenerateARPRequest` but from the function calling this?

File: arp.go [hunk 4]
```diff
@@ -271,32 +269,55 @@ func CreateARPRequest(gMAC []byte, dstIP string, switchIP string, arpTable *ArpT
...
-// CreateARPReply crafts an ARP Reply
-func CreateARPReply(srcAddr net.HardwareAddr, dstAddr net.HardwareAddr, spa net.IP, tpa net.IP) []byte {
-	frameBuf := make([]byte, 2000)
+// GenerateARPReply crafts an ARP Reply
+func GenerateARPReply(srcMAC net.HardwareAddr, dstMAC net.HardwareAddr, spa net.IP, tpa net.IP) []byte {
+
+	// Make space for nv header + ethernet header + ARP request
+	frameBuf := make([]byte, 4+14+28)
+
+	// nvHeader length value (the 2 bytes length field doesn't count so it's 2, not 4 byteshhh
```
Comment (si74) [inline arp.go:282]: is this a typo or an acronym that I'm missing? i.e. `byteshhh` @nicboul

File: arp.go (hunk 2 above)
Comment (nicboul) [reply arp.go:225]: because it's the length of the payload and the payload is everything after the length field

File: arp.go (hunk 3 above)
Comment (nicboul) [reply arp.go]: yeah for now I was thinking after or before calling GenerateARPRequest()

File: arp.go (hunk 4 above)
Comment (nicboul) [reply arp.go:282]: typo hehe, I meant 'bytes'

**Pull request merged**